cli/template_render.go

```diff
@@ -0,0 +1,139 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/prometheus/alertmanager/template"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+)
+
+var defaultData = template.Data{
+	Receiver: "receiver",
+	Status:   "alertstatus",
+	Alerts: template.Alerts{
+		template.Alert{
+			Status: "alertstatus",
+			Labels: template.KV{
+				"label1":          "value1",
+				"label2":          "value2",
+				"instance":        "foo.bar:1234",
+				"commonlabelkey1": "commonlabelvalue1",
+				"commonlabelkey2": "commonlabelvalue2",
+			},
+			Annotations: template.KV{
+				"annotation1":          "value1",
+				"annotation2":          "value2",
+				"commonannotationkey1": "commonannotationvalue1",
+				"commonannotationkey2": "commonannotationvalue2",
+			},
+			StartsAt:     time.Now().Add(-5 * time.Minute),
+			EndsAt:       time.Now(),
+			GeneratorURL: "https://generatorurl.com",
+			Fingerprint:  "fingerprint1",
+		},
+		template.Alert{
+			Status: "alertstatus",
+			Labels: template.KV{
+				"foo":             "bar",
+				"baz":             "qux",
+				"commonlabelkey1": "commonlabelvalue1",
+				"commonlabelkey2": "commonlabelvalue2",
+			},
+			Annotations: template.KV{
+				"aaa":                  "bbb",
+				"ccc":                  "ddd",
+				"commonannotationkey1": "commonannotationvalue1",
+				"commonannotationkey2": "commonannotationvalue2",
+			},
+			StartsAt:     time.Now().Add(-10 * time.Minute),
+			EndsAt:       time.Now(),
+			GeneratorURL: "https://generatorurl.com",
+			Fingerprint:  "fingerprint2",
+		},
+	},
+	GroupLabels: template.KV{
+		"grouplabelkey1": "grouplabelvalue1",
+		"grouplabelkey2": "grouplabelvalue2",
+	},
+	CommonLabels: template.KV{
+		"commonlabelkey1": "commonlabelvalue1",
+		"commonlabelkey2": "commonlabelvalue2",
+	},
+	CommonAnnotations: template.KV{
+		"commonannotationkey1": "commonannotationvalue1",
+		"commonannotationkey2": "commonannotationvalue2",
+	},
+	ExternalURL: "https://example.com",
+}
+
+type templateRenderCmd struct {
+	templateFilesGlobs []string
+	templateType       string
+	templateText       string
+	templateFile       *os.File
+}
+
+func configureTemplateRenderCmd(cc *kingpin.CmdClause) {
+	var (
+		c         = &templateRenderCmd{}
+		renderCmd = cc.Command("render", "Render a given definition in a template file to standard output")
+	)
+
+	renderCmd.Flag("templateglob", "Glob of paths that will be expanded and used for rendering").Required().StringsVar(&c.templateFilesGlobs)
+	renderCmd.Flag("templatetext", "The template that will be rendered").Required().StringVar(&c.templateText)
+	renderCmd.Flag("templatetype", "The type of the template. Can be either text (default) or html").EnumVar(&c.templateType, "html", "text")
```

Comment:
```suggestion
	renderCmd.Flag("template.type", "The type of the template. Can be either text (default) or html").EnumVar(&c.templateType, "html", "text")
```
Similar to https://github.com/prometheus/alertmanager/blob/4d7251fda16b0e97fe584ecac1306929da0a5a27/cli/routing.go#L56 (as @roidelapluie mentioned)